[Automation] java run fails for org.scalatest:scalatest-diagrams_3:3.3.0-SNAP4

## java run on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: java run
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.3.0-SNAP4" ./gradlew clean javaTest -Pcoordinates="org.scalatest:scalatest-diagrams_3:3.2.19"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/26036219276/job/76536247324
Last 300 lines of the log:
```console
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.scalatest/scalatest-diagrams_3/3.2.19/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 825ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: org.scalatest:scalatest-diagrams_3:3.2.19
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileScala NO-SOURCE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:jar UP-TO-DATE
> Task :compileTestJava NO-SOURCE

> Task :compileTestScala
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.scalatest/scalatest-diagrams_3/3.2.19/src/test/scala/org_scalatest/scalatest_diagrams_3/Scalatest_diagrams_3Test.scala:156:61: value byNameExpr is not a member of object org.scalatest.diagrams.DiagrammedExpr
one error found

> Task :compileTestScala FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.scalatest/scalatest-diagrams_3/3.2.19/build/reports/problems/problems-report.html


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestScala'.
> Compilation failed

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).
Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 5s
11 actionable tasks: 1 executed, 10 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.scalatest:scalatest-diagrams_3:3.2.19 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 7s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 3.3.0-SNAP4 at 2026-05-18T13:35:41Z with exit code 1 after 8s
❌ ERROR: 'native-image run' for 3.3.0-SNAP4 failed with exit code 1.
SOFT-FAIL[native-image run][3.3.0-SNAP4][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.3.0-SNAP4" ./gradlew clean test -Pcoordinates="org.scalatest:scalatest-diagrams_3:3.2.19"]
========================================================================================
 org.scalatest:scalatest-diagrams_3:3.2.19:3.3.0-SNAP4 stage javac compile
========================================================================================
Starting stage 'javac compile' for 3.3.0-SNAP4 at 2026-05-18T13:35:41Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.3.0-SNAP4" ./gradlew clean compileTestJava -Pcoordinates="org.scalatest:scalatest-diagrams_3:3.2.19"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.scalatest/scalatest-diagrams_3/3.2.19/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 772ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileScala NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :compileTestJava NO-SOURCE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.scalatest/scalatest-diagrams_3/3.2.19/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 789ms
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 3.3.0-SNAP4 at 2026-05-18T13:35:43Z with exit code 0 after 2s
========================================================================================
 org.scalatest:scalatest-diagrams_3:3.2.19:3.3.0-SNAP4 stage java run
========================================================================================
Starting stage 'java run' for 3.3.0-SNAP4 at 2026-05-18T13:35:43Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.3.0-SNAP4" ./gradlew clean javaTest -Pcoordinates="org.scalatest:scalatest-diagrams_3:3.2.19"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.scalatest/scalatest-diagrams_3/3.2.19/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 763ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :checkTestTimeoutAnnotations
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :tck-test-support:jar UP-TO-DATE
> Task :compileScala NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :compileTestJava NO-SOURCE

> Task :compileTestScala
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.scalatest/scalatest-diagrams_3/3.2.19/src/test/scala/org_scalatest/scalatest_diagrams_3/Scalatest_diagrams_3Test.scala:156:61: value byNameExpr is not a member of object org.scalatest.diagrams.DiagrammedExpr
one error found

> Task :compileTestScala FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.scalatest/scalatest-diagrams_3/3.2.19/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestScala'.
> Compilation failed

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 5s
12 actionable tasks: 2 executed, 10 up-to-date

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 7s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'java run' for 3.3.0-SNAP4 at 2026-05-18T13:35:51Z with exit code 1 after 8s
❌ ERROR: 'java run' for 3.3.0-SNAP4 failed with exit code 1.
FAILED[java run][3.3.0-SNAP4][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.3.0-SNAP4" ./gradlew clean javaTest -Pcoordinates="org.scalatest:scalatest-diagrams_3:3.2.19"]
```

## java run on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: java run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.3.0-SNAP4" ./gradlew clean javaTest -Pcoordinates="org.scalatest:scalatest-diagrams_3:3.2.19"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/26036219276/job/76536247286
Last 300 lines of the log:
```console
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.scalatest/scalatest-diagrams_3/3.2.19/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 881ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: org.scalatest:scalatest-diagrams_3:3.2.19
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :compileScala NO-SOURCE
> Task :processResources NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :classes UP-TO-DATE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :tck-test-support:jar UP-TO-DATE
> Task :jar
> Task :compileTestJava NO-SOURCE
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.scalatest/scalatest-diagrams_3/3.2.19/src/test/scala/org_scalatest/scalatest_diagrams_3/Scalatest_diagrams_3Test.scala:156:61: value byNameExpr is not a member of object org.scalatest.diagrams.DiagrammedExpr

> Task :compileTestScala
one error found

> Task :compileTestScala FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.scalatest/scalatest-diagrams_3/3.2.19/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

Execution failed for task ':compileTestScala'.
> Compilation failed

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).
You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
12 actionable tasks: 2 executed, 10 up-to-date

BUILD FAILED in 6s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.scalatest:scalatest-diagrams_3:3.2.19 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 8s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 3.3.0-SNAP4 at 2026-05-18T13:35:56Z with exit code 1 after 8s
❌ ERROR: 'native-image run' for 3.3.0-SNAP4 failed with exit code 1.
SOFT-FAIL[native-image run][3.3.0-SNAP4][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.3.0-SNAP4" ./gradlew clean test -Pcoordinates="org.scalatest:scalatest-diagrams_3:3.2.19"]
========================================================================================
 org.scalatest:scalatest-diagrams_3:3.2.19:3.3.0-SNAP4 stage javac compile
========================================================================================
Starting stage 'javac compile' for 3.3.0-SNAP4 at 2026-05-18T13:35:56Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.3.0-SNAP4" ./gradlew clean compileTestJava -Pcoordinates="org.scalatest:scalatest-diagrams_3:3.2.19"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.scalatest/scalatest-diagrams_3/3.2.19/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 826ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileScala NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :compileTestJava NO-SOURCE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.scalatest/scalatest-diagrams_3/3.2.19/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 818ms
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 3.3.0-SNAP4 at 2026-05-18T13:35:59Z with exit code 0 after 3s
========================================================================================
 org.scalatest:scalatest-diagrams_3:3.2.19:3.3.0-SNAP4 stage java run
========================================================================================
Starting stage 'java run' for 3.3.0-SNAP4 at 2026-05-18T13:35:59Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.3.0-SNAP4" ./gradlew clean javaTest -Pcoordinates="org.scalatest:scalatest-diagrams_3:3.2.19"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.scalatest/scalatest-diagrams_3/3.2.19/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 788ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :checkTestTimeoutAnnotations
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :compileScala NO-SOURCE
> Task :tck-test-support:jar UP-TO-DATE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :compileTestJava NO-SOURCE

> Task :compileTestScala
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.scalatest/scalatest-diagrams_3/3.2.19/src/test/scala/org_scalatest/scalatest_diagrams_3/Scalatest_diagrams_3Test.scala:156:61: value byNameExpr is not a member of object org.scalatest.diagrams.DiagrammedExpr
one error found

> Task :compileTestScala FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.scalatest/scalatest-diagrams_3/3.2.19/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.


Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

* What went wrong:
Execution failed for task ':compileTestScala'.
> Compilation failed

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).
You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 5s
12 actionable tasks: 2 executed, 10 up-to-date

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 7s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'java run' for 3.3.0-SNAP4 at 2026-05-18T13:36:07Z with exit code 1 after 8s
❌ ERROR: 'java run' for 3.3.0-SNAP4 failed with exit code 1.
FAILED[java run][3.3.0-SNAP4][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.3.0-SNAP4" ./gradlew clean javaTest -Pcoordinates="org.scalatest:scalatest-diagrams_3:3.2.19"]
```

## java run on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: java run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="3.3.0-SNAP4" ./gradlew clean javaTest -Pcoordinates="org.scalatest:scalatest-diagrams_3:3.2.19"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/26036219276/job/76536247521
Last 300 lines of the log:
```console
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.scalatest/scalatest-diagrams_3/3.2.19/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 827ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: org.scalatest:scalatest-diagrams_3:3.2.19
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileScala NO-SOURCE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:jar UP-TO-DATE
> Task :jar
> Task :compileTestJava NO-SOURCE

> Task :compileTestScala
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.scalatest/scalatest-diagrams_3/3.2.19/src/test/scala/org_scalatest/scalatest_diagrams_3/Scalatest_diagrams_3Test.scala:156:61: value byNameExpr is not a member of object org.scalatest.diagrams.DiagrammedExpr
one error found

> Task :compileTestScala FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.scalatest/scalatest-diagrams_3/3.2.19/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

Execution failed for task ':compileTestScala'.
> Compilation failed

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 5s
For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
12 actionable tasks: 2 executed, 10 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.scalatest:scalatest-diagrams_3:3.2.19 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 7s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 3.3.0-SNAP4 at 2026-05-18T13:35:58Z with exit code 1 after 8s
❌ ERROR: 'native-image run' for 3.3.0-SNAP4 failed with exit code 1.
SOFT-FAIL[native-image run][3.3.0-SNAP4][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="3.3.0-SNAP4" ./gradlew clean test -Pcoordinates="org.scalatest:scalatest-diagrams_3:3.2.19"]
========================================================================================
 org.scalatest:scalatest-diagrams_3:3.2.19:3.3.0-SNAP4 stage javac compile
========================================================================================
Starting stage 'javac compile' for 3.3.0-SNAP4 at 2026-05-18T13:35:58Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="3.3.0-SNAP4" ./gradlew clean compileTestJava -Pcoordinates="org.scalatest:scalatest-diagrams_3:3.2.19"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.scalatest/scalatest-diagrams_3/3.2.19/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 779ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileScala NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :compileTestJava NO-SOURCE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.scalatest/scalatest-diagrams_3/3.2.19/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 797ms
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 3.3.0-SNAP4 at 2026-05-18T13:36:01Z with exit code 0 after 3s
========================================================================================
 org.scalatest:scalatest-diagrams_3:3.2.19:3.3.0-SNAP4 stage java run
========================================================================================
Starting stage 'java run' for 3.3.0-SNAP4 at 2026-05-18T13:36:01Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="3.3.0-SNAP4" ./gradlew clean javaTest -Pcoordinates="org.scalatest:scalatest-diagrams_3:3.2.19"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.scalatest/scalatest-diagrams_3/3.2.19/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 790ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :checkTestTimeoutAnnotations
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :compileScala NO-SOURCE
> Task :tck-test-support:jar UP-TO-DATE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :compileTestJava NO-SOURCE

> Task :compileTestScala
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.scalatest/scalatest-diagrams_3/3.2.19/src/test/scala/org_scalatest/scalatest_diagrams_3/Scalatest_diagrams_3Test.scala:156:61: value byNameExpr is not a member of object org.scalatest.diagrams.DiagrammedExpr

> Task :compileTestScala FAILED
one error found

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.scalatest/scalatest-diagrams_3/3.2.19/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestScala'.
> Compilation failed

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 5s
12 actionable tasks: 2 executed, 10 up-to-date

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 7s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'java run' for 3.3.0-SNAP4 at 2026-05-18T13:36:08Z with exit code 1 after 7s
❌ ERROR: 'java run' for 3.3.0-SNAP4 failed with exit code 1.
FAILED[java run][3.3.0-SNAP4][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="3.3.0-SNAP4" ./gradlew clean javaTest -Pcoordinates="org.scalatest:scalatest-diagrams_3:3.2.19"]
```

--- a/metadata/org.scalatest/scalatest-diagrams_3/3.3.0-SNAP4/reachability-metadata.json
+++ b/metadata/org.scalatest/scalatest-diagrams_3/3.3.0-SNAP4/reachability-metadata.json
@@ -1,0 +1,78 @@
+{
+  "reflection": [
+    {
+      "condition": {
+        "typeReached": "org.scalatest.Assertions"
+      },
+      "type": "org.scalatest.Assertions$",
+      "fields": [
+        {
+          "name": "0bitmap$1"
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "org.scalatest.diagrams.Diagrams"
+      },
+      "type": "org.scalatest.diagrams.Diagrams$",
+      "fields": [
+        {
+          "name": "0bitmap$1"
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "org.scalactic.source.Position"
+      },
+      "type": "org.scalactic.source.Position$",
+      "fields": [
+        {
+          "name": "0bitmap$1"
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "org.scalatest.diagrams.DiagrammedByNameExpr"
+      },
+      "type": "org.scalatest.diagrams.DiagrammedByNameExpr",
+      "fields": [
+        {
+          "name": "0bitmap$1"
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "org.scalatest.Resources$"
+      },
+      "type": "org.scalatest.Resources$",
+      "fields": [
+        {
+          "name": "0bitmap$1"
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "org.scalatest.exceptions.StackDepthException"
+      },
+      "type": "org.scalatest.exceptions.StackDepthException",
+      "fields": [
+        {
+          "name": "0bitmap$1"
+        }
+      ]
+    }
+  ],
+  "resources": [
+    {
+      "condition": {
+        "typeReached": "org.scalatest.Resources$"
+      },
+      "bundle": "org.scalatest.ScalaTestBundle"
+    }
+  ]
+}

--- a/metadata/org.scalatest/scalatest-diagrams_3/index.json
+++ b/metadata/org.scalatest/scalatest-diagrams_3/index.json
@@ -23,5 +23,21 @@
       "org.scalatest",
       "org.scalatest.exceptions"
     ]
+  },
+  {
+    "metadata-version" : "3.3.0-SNAP4",
+    "language" : {
+      "name" : "scala",
+      "version" : "3"
+    },
+    "tested-versions" : [
+      "3.3.0-SNAP4"
+    ],
+    "allowed-packages" : [
+      "org.scalatest.diagrams",
+      "org.scalactic.source",
+      "org.scalatest",
+      "org.scalatest.exceptions"
+    ]
   }
 ]

--- a/metadata/org.scalatest/scalatest-diagrams_3/index.json
+++ b/metadata/org.scalatest/scalatest-diagrams_3/index.json
@@ -26,6 +26,11 @@
   },
   {
     "metadata-version" : "3.3.0-SNAP4",
+    "source-code-url" : "https://repo.maven.apache.org/maven2/org/scalatest/scalatest-diagrams_3/$version$/scalatest-diagrams_3-$version$-sources.jar",
+    "repository-url" : "https://github.com/scalatest/scalatest",
+    "test-code-url" : "N/A",
+    "documentation-url" : "https://repo.maven.apache.org/maven2/org/scalatest/scalatest-diagrams_3/$version$/scalatest-diagrams_3-$version$-javadoc.jar",
+    "description" : "ScalaTest Diagrams adds diagrammed assertion support to ScalaTest, showing evaluated subexpressions when an assertion fails. The Scala 3 artifact provides the org.scalatest.diagrams APIs and macros for writing ScalaTest assertions whose failure messages include expression diagrams.",
     "language" : {
       "name" : "scala",
       "version" : "3"

--- a/stats/org.scalatest/scalatest-diagrams_3/3.3.0-SNAP4/execution-metrics.json
+++ b/stats/org.scalatest/scalatest-diagrams_3/3.3.0-SNAP4/execution-metrics.json
@@ -1,0 +1,87 @@
+{
+  "fix_java_run_fail:2026-06-06": {
+    "timestamp": "2026-06-06T20:41:03.004230Z",
+    "library": "org.scalatest:scalatest-diagrams_3:3.3.0-SNAP4",
+    "starting_commit": "fdf9ed0dd48a12a50fb082f96f9c05d9a023c62a",
+    "ending_commit": "00e39c1f12b4f877291b95e200bc311c648c3cdc",
+    "previous_library": "org.scalatest:scalatest-diagrams_3:3.2.19",
+    "strategy_name": "java_run_iterative_with_coverage_sources_pi_gpt-5.5",
+    "agent": "pi",
+    "model": "oca/gpt-5.5",
+    "status": "success",
+    "stats": {
+      "version": "3.3.0-SNAP4",
+      "dynamicAccess": {
+        "breakdown": {},
+        "coverageRatio": 1.0,
+        "coveredCalls": 0,
+        "totalCalls": 0
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 623,
+          "missed": 3019,
+          "ratio": 0.17106,
+          "total": 3642
+        },
+        "line": {
+          "covered": 63,
+          "missed": 128,
+          "ratio": 0.329843,
+          "total": 191
+        },
+        "method": {
+          "covered": 47,
+          "missed": 82,
+          "ratio": 0.364341,
+          "total": 129
+        }
+      }
+    },
+    "previous_library_stats": {
+      "version": "3.2.19",
+      "dynamicAccess": {
+        "breakdown": {},
+        "coverageRatio": 1.0,
+        "coveredCalls": 0,
+        "totalCalls": 0
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 698,
+          "missed": 3185,
+          "ratio": 0.179758,
+          "total": 3883
+        },
+        "line": {
+          "covered": 67,
+          "missed": 130,
+          "ratio": 0.340102,
+          "total": 197
+        },
+        "method": {
+          "covered": 53,
+          "missed": 89,
+          "ratio": 0.373239,
+          "total": 142
+        }
+      }
+    },
+    "metrics": {
+      "input_tokens_used": 28758,
+      "cached_input_tokens_used": 306176,
+      "output_tokens_used": 2812,
+      "iterations": 1,
+      "cost_usd": 0.2375,
+      "tested_library_loc": 63,
+      "metadata_entries": 13,
+      "previous_library_metadata_entries": 13,
+      "code_coverage_percent": 32.98,
+      "previous_library_coverage_percent": 34.01
+    },
+    "artifacts": {
+      "test_file": "tests/src/org.scalatest/scalatest-diagrams_3/3.3.0-SNAP4/src/test/scala/org_scalatest/scalatest_diagrams_3/Scalatest_diagrams_3Test.scala",
+      "metadata_file": "metadata/org.scalatest/scalatest-diagrams_3/3.3.0-SNAP4/reachability-metadata.json"
+    }
+  }
+}

--- a/stats/org.scalatest/scalatest-diagrams_3/3.3.0-SNAP4/stats.json
+++ b/stats/org.scalatest/scalatest-diagrams_3/3.3.0-SNAP4/stats.json
@@ -1,0 +1,31 @@
+{
+  "versions" : [ {
+    "version" : "3.3.0-SNAP4",
+    "dynamicAccess" : {
+      "breakdown" : { },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 0,
+      "totalCalls" : 0
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 623,
+        "missed" : 3019,
+        "ratio" : 0.17106,
+        "total" : 3642
+      },
+      "line" : {
+        "covered" : 63,
+        "missed" : 128,
+        "ratio" : 0.329843,
+        "total" : 191
+      },
+      "method" : {
+        "covered" : 47,
+        "missed" : 82,
+        "ratio" : 0.364341,
+        "total" : 129
+      }
+    }
+  } ]
+}

--- a/tests/src/org.scalatest/scalatest-diagrams_3/3.3.0-SNAP4/.gitignore
+++ b/tests/src/org.scalatest/scalatest-diagrams_3/3.3.0-SNAP4/.gitignore
@@ -1,0 +1,4 @@
+gradlew.bat
+gradlew
+gradle/
+build/

--- a/tests/src/org.scalatest/scalatest-diagrams_3/3.3.0-SNAP4/build.gradle
+++ b/tests/src/org.scalatest/scalatest-diagrams_3/3.3.0-SNAP4/build.gradle
@@ -1,0 +1,36 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+plugins {
+    id "org.graalvm.internal.tck"
+    id "scala"
+}
+
+String libraryVersion = tck.testedLibraryVersion.get()
+
+dependencies {
+    testImplementation "org.scalatest:scalatest-diagrams_3:$libraryVersion"
+    testImplementation 'org.assertj:assertj-core:3.22.0'
+    testImplementation 'org.scala-lang:scala3-library_3:3.3.6'
+    testImplementation 'org.scala-lang:scala3-compiler_3:3.3.6'
+}
+
+java {
+    toolchain {
+        languageVersion = JavaLanguageVersion.of(21)
+    }
+}
+
+graalvmNative {
+    agent {
+        defaultMode = "conditional"
+        modes {
+            conditional {
+                userCodeFilterPath = "user-code-filter.json"
+            }
+        }
+    }
+}

--- a/tests/src/org.scalatest/scalatest-diagrams_3/3.3.0-SNAP4/gradle.properties
+++ b/tests/src/org.scalatest/scalatest-diagrams_3/3.3.0-SNAP4/gradle.properties
@@ -1,0 +1,6 @@
+# Optional explicit tested library coordinates (format: group:artifact:version).
+# If omitted, fallback resolution in org.graalvm.internal.tck.gradle still uses
+# environment variable GVM_TCK_LC and then this property.
+library.coordinates = org.scalatest:scalatest-diagrams_3:3.3.0-SNAP4
+library.version = 3.3.0-SNAP4
+metadata.dir = org.scalatest/scalatest-diagrams_3/3.3.0-SNAP4/

--- a/tests/src/org.scalatest/scalatest-diagrams_3/3.3.0-SNAP4/settings.gradle
+++ b/tests/src/org.scalatest/scalatest-diagrams_3/3.3.0-SNAP4/settings.gradle
@@ -1,0 +1,20 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+pluginManagement {
+    def tckPath = Objects.requireNonNullElse(
+            System.getenv("GVM_TCK_TCKDIR"),
+            "../../../../tck-build-logic"
+    )
+    includeBuild(tckPath)
+    repositories { mavenLocal(); gradlePluginPortal() }
+}
+
+plugins {
+    id "org.graalvm.internal.tck-settings" version "1.0.0-SNAPSHOT"
+}
+
+rootProject.name = 'org.scalatest.scalatest-diagrams_3_tests'

--- a/tests/src/org.scalatest/scalatest-diagrams_3/3.3.0-SNAP4/src/test/scala/org_scalatest/scalatest_diagrams_3/Scalatest_diagrams_3Test.scala
+++ b/tests/src/org.scalatest/scalatest-diagrams_3/3.3.0-SNAP4/src/test/scala/org_scalatest/scalatest_diagrams_3/Scalatest_diagrams_3Test.scala
@@ -151,22 +151,16 @@ class Scalatest_diagrams_3Test extends Diagrams:
     assertFalse(anchors.exists(_.contains("ignored")), anchors.toString)
 
   @Test
-  def byNameDiagrammedExpressionEvaluatesValueLazilyAndOnlyOnce(): Unit =
-    var evaluations: Int = 0
-    val expression: DiagrammedExpr[String] = DiagrammedExpr.byNameExpr(
-      {
-        evaluations += 1
-        "computed-value"
-      },
-      31
-    )
+  def diagrammedApplyExpressionKeepsTheLastValueForDuplicateQualifierAnchors(): Unit =
+    val base: DiagrammedExpr[String] = DiagrammedExpr.simpleExpr("base", 6)
+    val selectedWithSameAnchor: DiagrammedExpr[String] = DiagrammedExpr.selectExpr(base, "selected", 6)
 
-    assertEquals(31, expression.anchor)
-    assertTrue(expression.anchorValues.isEmpty)
-    assertEquals(0, evaluations)
-    assertEquals("computed-value", expression.value)
-    assertEquals("computed-value", expression.value)
-    assertEquals(1, evaluations)
+    val applied: DiagrammedExpr[String] = DiagrammedExpr.applyExpr(selectedWithSameAnchor, Nil, "result", 10)
+    val anchors: List[String] = applied.anchorValues.map(_.toString)
+
+    assertFalse(anchors.contains("AnchorValue(6,base)"), anchors.toString)
+    assertTrue(anchors.contains("AnchorValue(6,selected)"), anchors.toString)
+    assertTrue(anchors.contains("AnchorValue(10,result)"), anchors.toString)
 
   @Test
   def useDiagramMarkerCanBeObtainedFromTrait(): Unit =

--- a/tests/src/org.scalatest/scalatest-diagrams_3/3.3.0-SNAP4/src/test/scala/org_scalatest/scalatest_diagrams_3/Scalatest_diagrams_3Test.scala
+++ b/tests/src/org.scalatest/scalatest-diagrams_3/3.3.0-SNAP4/src/test/scala/org_scalatest/scalatest_diagrams_3/Scalatest_diagrams_3Test.scala
@@ -1,0 +1,190 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_scalatest.scalatest_diagrams_3
+
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertFalse
+import org.junit.jupiter.api.Assertions.assertNotNull
+import org.junit.jupiter.api.Assertions.assertThrows as junitAssertThrows
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.function.Executable
+import org.scalatest.diagrams.DiagrammedExpr
+import org.scalatest.diagrams.Diagrams
+import org.scalatest.exceptions.TestCanceledException
+import org.scalatest.exceptions.TestFailedException
+
+class Scalatest_diagrams_3Test extends Diagrams:
+  @Test
+  def passingDiagramAssertionsEvaluateOrdinaryBooleanExpressions(): Unit =
+    val numbers: Vector[Int] = Vector(2, 3, 5, 7)
+    val label: String = "reachability-metadata"
+    val calculator: Calculator = Calculator(40)
+
+    assert(numbers.sum == 17)
+    assert(numbers.filter(_ > 3).contains(7))
+    assert(label.startsWith("reachability") && label.endsWith("metadata"))
+    assert(calculator.incrementedBy(2) == 42, "calculator should add the requested delta")
+
+  @Test
+  def failingDiagramAssertionIncludesExpressionSourceAndCapturedValues(): Unit =
+    val numbers: Vector[Int] = Vector(1, 2, 3)
+    val requested: Int = 4
+
+    val failure: TestFailedException = expectTestFailure {
+      assert(numbers.contains(requested))
+    }
+
+    val message: String = failure.getMessage
+    assertTrue(message.contains("numbers.contains(requested)"), message)
+    assertTrue(message.contains("Vector(1, 2, 3)"), message)
+    assertTrue(message.contains("4"), message)
+    assertTrue(message.contains("false"), message)
+    assertTrue(message.contains("|"), message)
+
+  @Test
+  def failingDiagramAssertionWithCluePreservesTheClueAndDiagramDetails(): Unit =
+    val availableItems: Int = 2
+    val requestedItems: Int = 5
+    val clue: String = "inventory request should fit the available stock"
+
+    val failure: TestFailedException = expectTestFailure {
+      assert(requestedItems <= availableItems, clue)
+    }
+
+    val message: String = failure.getMessage
+    assertTrue(message.contains(clue), message)
+    assertTrue(message.contains("requestedItems <= availableItems"), message)
+    assertTrue(message.contains("5"), message)
+    assertTrue(message.contains("2"), message)
+    assertTrue(message.contains("false"), message)
+    assertTrue(message.contains("|"), message)
+
+  @Test
+  def failingDiagramAssertionRendersNestedSelectionsAndMethodApplications(): Unit =
+    val calculator: Calculator = Calculator(6)
+    val delta: Int = 5
+    val expected: Int = 99
+
+    val failure: TestFailedException = expectTestFailure {
+      assert(calculator.incrementedBy(delta) == expected)
+    }
+
+    val message: String = failure.getMessage
+    assertTrue(message.contains("calculator.incrementedBy(delta) == expected"), message)
+    assertTrue(message.contains("Calculator(6)"), message)
+    assertTrue(message.contains("11"), message)
+    assertTrue(message.contains("5"), message)
+    assertTrue(message.contains("99"), message)
+    assertTrue(message.contains("false"), message)
+
+  @Test
+  def failingTripleEqualsDiagramAssertionRendersComparedValues(): Unit =
+    val actual: Int = 3
+    val expected: Int = 5
+
+    assert(actual === 3)
+
+    val failure: TestFailedException = expectTestFailure {
+      assert(actual === expected)
+    }
+
+    val message: String = failure.getMessage
+    assertTrue(message.contains("actual === expected"), message)
+    assertTrue(message.contains("3"), message)
+    assertTrue(message.contains("5"), message)
+    assertTrue(message.contains("false"), message)
+    assertTrue(message.contains("|"), message)
+
+  @Test
+  def failingDiagramAssumeThrowsCanceledExceptionWithDiagramDetails(): Unit =
+    val serviceName: String = "metadata-service"
+    val requestedPrefix: String = "native-image"
+
+    val cancellation: TestCanceledException = expectTestCancellation {
+      assume(serviceName.startsWith(requestedPrefix))
+    }
+
+    val message: String = cancellation.getMessage
+    assertTrue(message.contains("serviceName.startsWith(requestedPrefix)"), message)
+    assertTrue(message.contains("metadata-service"), message)
+    assertTrue(message.contains("native-image"), message)
+    assertTrue(message.contains("false"), message)
+
+  @Test
+  def diagrammedSimpleAndSelectExpressionsExposeAnchorValues(): Unit =
+    val base: DiagrammedExpr[String] = DiagrammedExpr.simpleExpr("metadata", 2)
+    val selected: DiagrammedExpr[Int] = DiagrammedExpr.selectExpr(base, 8, 12)
+
+    assertEquals("metadata", base.value)
+    assertEquals(2, base.anchor)
+    assertEquals(List("AnchorValue(2,metadata)"), base.anchorValues.map(_.toString))
+
+    assertEquals(8, selected.value)
+    assertEquals(12, selected.anchor)
+    assertEquals(
+      List("AnchorValue(2,metadata)", "AnchorValue(12,8)"),
+      selected.anchorValues.map(_.toString)
+    )
+
+  @Test
+  def diagrammedApplyExpressionCombinesQualifierResultAndArguments(): Unit =
+    val qualifier: DiagrammedExpr[String] = DiagrammedExpr.simpleExpr("scala", 1)
+    val firstArgument: DiagrammedExpr[Int] = DiagrammedExpr.simpleExpr(3, 9)
+    val secondArgument: DiagrammedExpr[Int] = DiagrammedExpr.simpleExpr(4, 14)
+    val syntheticArgument: DiagrammedExpr[String] = DiagrammedExpr.simpleExpr("ignored", -1)
+    val arguments: List[DiagrammedExpr[?]] = List(firstArgument, secondArgument, syntheticArgument)
+
+    val applied: DiagrammedExpr[String] = DiagrammedExpr.applyExpr(qualifier, arguments, "scala-7", 20)
+    val anchors: List[String] = applied.anchorValues.map(_.toString)
+
+    assertEquals("scala-7", applied.value)
+    assertEquals(20, applied.anchor)
+    assertEquals("AnchorValue(1,scala)", anchors.head)
+    assertTrue(anchors.contains("AnchorValue(20,scala-7)"), anchors.toString)
+    assertTrue(anchors.contains("AnchorValue(9,3)"), anchors.toString)
+    assertTrue(anchors.contains("AnchorValue(14,4)"), anchors.toString)
+    assertFalse(anchors.exists(_.contains("ignored")), anchors.toString)
+
+  @Test
+  def byNameDiagrammedExpressionEvaluatesValueLazilyAndOnlyOnce(): Unit =
+    var evaluations: Int = 0
+    val expression: DiagrammedExpr[String] = DiagrammedExpr.byNameExpr(
+      {
+        evaluations += 1
+        "computed-value"
+      },
+      31
+    )
+
+    assertEquals(31, expression.anchor)
+    assertTrue(expression.anchorValues.isEmpty)
+    assertEquals(0, evaluations)
+    assertEquals("computed-value", expression.value)
+    assertEquals("computed-value", expression.value)
+    assertEquals(1, evaluations)
+
+  @Test
+  def useDiagramMarkerCanBeObtainedFromTrait(): Unit =
+    assertNotNull(UseDiagram)
+
+  private def expectTestFailure(block: => Unit): TestFailedException =
+    junitAssertThrows(
+      classOf[TestFailedException],
+      new Executable:
+        override def execute(): Unit = block
+    )
+
+  private def expectTestCancellation(block: => Unit): TestCanceledException =
+    junitAssertThrows(
+      classOf[TestCanceledException],
+      new Executable:
+        override def execute(): Unit = block
+    )
+
+  private final case class Calculator(base: Int):
+    def incrementedBy(delta: Int): Int = base + delta

--- a/tests/src/org.scalatest/scalatest-diagrams_3/3.3.0-SNAP4/user-code-filter.json
+++ b/tests/src/org.scalatest/scalatest-diagrams_3/3.3.0-SNAP4/user-code-filter.json
@@ -7,6 +7,15 @@
       "includeClasses" : "org.scalatest.diagrams.**"
     },
     {
+      "includeClasses" : "org.scalactic.source.**"
+    },
+    {
+      "includeClasses" : "org.scalatest.**"
+    },
+    {
+      "includeClasses" : "org.scalatest.exceptions.**"
+    },
+    {
       "includeClasses" : "org_scalatest.scalatest_diagrams_3.**"
     }
   ]

--- a/tests/src/org.scalatest/scalatest-diagrams_3/3.3.0-SNAP4/user-code-filter.json
+++ b/tests/src/org.scalatest/scalatest-diagrams_3/3.3.0-SNAP4/user-code-filter.json
@@ -1,0 +1,13 @@
+{
+  "rules" : [
+    {
+      "excludeClasses" : "**"
+    },
+    {
+      "includeClasses" : "org.scalatest.diagrams.**"
+    },
+    {
+      "includeClasses" : "org_scalatest.scalatest_diagrams_3.**"
+    }
+  ]
+}

